src/system.rs

```diff
@@ -31,3 +127,88 @@ mod unit_tests {
         assert_eq!(uptime_internal("12489513.08 22906637.29\n"), 12489513);
     }
 }
+
+/// Returns information about virtual memory usage
+///
+/// `/proc/meminfo` contains the virtual memory statistics
+pub fn virtual_memory() -> Option<VirtualMemory> {
+    let data = read_file(Path::new("/proc/meminfo")).unwrap();
+    let mem_info = make_map(&data);
+
+    let mut virtual_memory: VirtualMemory = Default::default();
+
+    virtual_memory.total = get_deref!(mem_info, "MemTotal:");
```

Comment:
If you make this return Result you can probably do:

```
mem_info.get("MemTotal:").ok_or("MemTotal field not found")?
```

(get returns an Option, ok_or converts that to a Result, and ? does an early return on error)

That means you can get rid of get_deref in favour of something more standard and idiomatic.